### PR TITLE
Fix `nerdctl build` logs error even when succeeds

### PR DESCRIPTION
**Pull request by ktock**
Nerdctl prints error logs even when buildkitd is running on the host and even when nerdctl successfully detects it.
This commit fixes nerdctl to print logs only when it fails to detect buildkitd sockets.

### When buildkitd is running

Now nerdctl doesn't print false error logs.

```console
$ nerdctl --debug build -t foo /tmp/ctx
DEBU[0000] stateDir: /run/user/1000/containerd-rootless 
DEBU[0000] rootless parent main: executing "/usr/bin/nsenter" with [-r/ -w/home/rootless --preserve-credentials -m -n -U -t 267 -F nerdctl --debug build -t foo /tmp/ctx] 
DEBU[0000] worker labels: map[org.mobyproject.buildkit.worker.executor:oci org.mobyproject.buildkit.worker.hostname:4398f021caaf org.mobyproject.buildkit.worker.network:host org.mobyproject.buildkit.worker.oci.process-mode:sandbox org.mobyproject.buildkit.worker.snapshotter:overlayfs] 
DEBU[0000] running /usr/local/bin/buildctl [--addr=unix:///run/user/1000/buildkit/buildkitd.sock build --progress=auto --frontend=dockerfile.v0 --local=context=/tmp/ctx --local=dockerfile=/tmp/ctx --output=type=docker,name=docker.io/library/foo:latest] 
```

### When buildkitd is not running

nerdctl prints error logs with hints.

```console
$ nerdctl --debug build -t foo /tmp/ctx
nerdctl --debug build -t foo /tmp/ctx
DEBU[0000] stateDir: /run/user/1000/containerd-rootless 
DEBU[0000] rootless parent main: executing "/usr/bin/nsenter" with [-r/ -w/home/rootless --preserve-credentials -m -n -U -t 267 -F nerdctl --debug build -t foo /tmp/ctx] 
ERRO[0000] `buildctl` needs to be installed and `buildkitd` needs to be running, see https://github.com/moby/buildkit , and `containerd-rootless-setuptool.sh install-buildkit` for OCI worker or `containerd-rootless-setuptool.sh install-buildkit-containerd` for containerd worker  error="2 errors occurred:\n\t* failed to ping to host unix:///run/user/1000/buildkit-default/buildkitd.sock: exit status 1\n\t* failed to ping to host unix:///run/user/1000/buildkit/buildkitd.sock: exit status 1\n\n"
FATA[0000] no buildkit host is available: 2 errors occurred:
	* failed to ping to host unix:///run/user/1000/buildkit-default/buildkitd.sock: exit status 1
	* failed to ping to host unix:///run/user/1000/buildkit/buildkitd.sock: exit status 1
```

### CODE REVIEW

**Comment by ktock** [comment]
@AkihiroSuda 
Does this fix https://github.com/containerd/nerdctl/issues/919 ?

File: pkg/buildkitutil/buildkitutil.go [hunk 1]
```diff
@@ -65,15 +65,17 @@ func GetBuildkitHost(namespace string) (string, error) {
 		hostRel = append(hostRel, fmt.Sprintf("buildkit-%s/buildkitd.sock", namespace))
 	}
 	hostRel = append(hostRel, "buildkit-default/buildkitd.sock", "buildkit/buildkitd.sock")
```
Comment (AkihiroSuda) [inline pkg/buildkitutil/buildkitutil.go:67]: logrus.Debugf("Choosing the buildkit host, candidates=%v", hostRel)

File: pkg/buildkitutil/buildkitutil.go [hunk 2]
```diff
@@ -65,15 +65,17 @@ func GetBuildkitHost(namespace string) (string, error) {
...
+		_, err := pingBKDaemon(buildkitHost)
 		if err == nil {
 			return buildkitHost, nil
 		}
-		logrus.WithField("host", buildkitHost).Warn(err)
+		allErr = multierror.Append(allErr, fmt.Errorf("failed to ping to host %s: %w", buildkitHost, err))
 	}
-	return "", fmt.Errorf("no buildkit host is available")
+	logrus.WithError(allErr).Error(getHint())
+	return "", fmt.Errorf("no buildkit host is available: %w", allErr)
```
Comment (AkihiroSuda) [inline pkg/buildkitutil/buildkitutil.go:78]:
```suggestion
	return "", fmt.Errorf("no buildkit host is available, tried %d candidates: %w", len(hostRel), allErr)
```

File: pkg/buildkitutil/buildkitutil.go [hunk 3]
```diff
@@ -65,15 +65,19 @@ func GetBuildkitHost(namespace string) (string, error) {
 		hostRel = append(hostRel, fmt.Sprintf("buildkit-%s/buildkitd.sock", namespace))
 	}
 	hostRel = append(hostRel, "buildkit-default/buildkitd.sock", "buildkit/buildkitd.sock")
+	var allErr error
 	for _, p := range hostRel {
+		logrus.Debugf("Choosing the buildkit host, candidates=%v (in %q)", hostRel, run)
```
Comment (AkihiroSuda) [inline pkg/buildkitutil/buildkitutil.go:70]: This is accidentally printed multiple times

**Review by AkihiroSuda — APPROVED** [review]
Thanks

**Pull request merged**